docs/apis-tools/java-client/job-worker.md

```diff
@@ -177,30 +173,16 @@ Even with streaming enabled, job workers still occasionally poll the cluster for
 
 This ensures polling should not activate any new jobs, and the worker will back off and poll less often as long as it receives empty responses overtime.
 
-### Caveats
-
-While job streaming is already usable, it is still a feature under development, and there are some major known caveats should you decide to use it:
-
-#### Lack of flow control
-
-There is currently no flow control mechanism between the workers and a Zeebe cluster. As Zeebe has no knowledge of the load on your workers, it's possible for a cluster to push too many jobs to your workers such that they become overloaded, and in the worst case, crash. That said, Zeebe will try to distribute the load across your workers evenly, but this is not guaranteed, and even if it were so, there is no guarantee each worker will perform in exactly the same way.
+#### Backpressure
 
-To avoid this, ensure you enable the job worker metrics for all your workers and monitor them closely. This would allow you to scale your worker deployment in or out, depending on the current load per worker.
+To avoid your workers being overloaded with too many jobs, e.g. running out of memory, the Java job worker relies on the [built-in gRPC flow control mechanism](https://grpc.io/docs/guides/flow-control/). If streaming is enabled, this means the worker will never work on more jobs than the configured `maxJobsActive` parameter. For example, if `maxJobsActive = 32`, then your worker will only work on at most 32 jobs concurrently. If this is already the case, and a 33rd job comes in, the gRPC thread will block, thus signaling the gateway to stop sending more jobs.
 
-Additionally, ensure workers perform tasks in a roughly similar time frame to avoid one or more workers becoming overwhelmed.
+**If streaming is enabled, back pressure applies to both pushing and polling**. You can then use `maxJobsActive` as a way to soft-bound the memory usage of your worker. For example, if your max message size is 4MB, and `maxJobsActive = 32`, then a single worker could use up to 128MB of memory in the worst case.
 
 :::note
-We're aware this is a large feature gap, and hope to address it soon.
+If the worker blocks longer than the job's deadline, the job will **not** be passed to the worker, but will be dropped. As it will time out on the broker side, it will be pushed again.
```

Comment:
```suggestion
If the worker blocks longer than the job's deadline, the job will **not** be passed to the worker, but will be dropped. As it will time out on the broker side, it will be polled again.
```

It is not pushed again or?

Reply:
On time out it's pushed again, though it might go to another worker. Only on YIELD it isn't pushed out.